File: my-folder/0121-best-time-to-buy-and-sell-stock/solution.java
class Solution {
    public int maxProfit(int[] prices) {
        int minPurchase = prices[0];
        int maxProfit = 0;
        for (int i = 1; i < prices.length; i++) {
            maxProfit = Math.max(maxProfit, prices[i] - minPurchase);
            minPurchase = Math.min(minPurchase, prices[i]);
        }
        return maxProfit;
    }
}

File: my-folder/0121-best-time-to-buy-and-sell-stock/solution.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        res = 0
        minPurchase = prices[0]

        for i in range(1, len(prices)):
            res = max(res, prices[i] - minPurchase)
            minPurchase = min(minPurchase, prices[i])
        
        return res
